core/common/src/grpc_interceptor.rs

```diff
@@ -154,19 +155,41 @@ where
         let interceptor = self.interceptor.clone();
 
         let (service_name, method_name) = Self::retrieve_grpc_names_from_uri(request.uri());
-        let is_applicable = interceptor.is_applicable(&service_name, &method_name);
+        let is_applicable = interceptor.is_applicable(&service_name, &method_name)
+            && (request.method() == Method::POST);
 
         if is_applicable && interceptor.must_handle_request() {
             let (parts, body) = request.into_parts();
-            let mut body_bytes: Bytes =
-                match futures::executor::block_on(hyper::body::to_bytes(body)) {
-                    Ok(bytes) => bytes,
-                    Err(err) => {
-                        return Box::pin(async move {
-                            Err(Box::new(err) as Box<dyn std::error::Error + Sync + Send>)
-                        })
-                    }
-                };
+
+            // Rust requires that we initilaize body_bytes_timeout_result.
+            // Note: We will never use the initilaized value, so using an Ok value is fine.
+            let mut body_bytes_timeout_result = Ok(Ok(Bytes::new()));
+
+            // There is a known issue where hyper::body::to_bytes sometimes hangs in the code below.
+            // We will use a timeout to break out when this happens.  This fix is a bandaid.  We will
+            // implement a better fix after we have upgraded to the latest major version of the hyper crate.
+            futures::executor::block_on(async {
```

Comment:
Do not know if this will work, but would be cleaner if so:

```
let body_bytes_timeout_result = futures::executor::block_on(async {
    async_std::future::timeout(
        core::time::Duration::from_secs(5),
        hyper::body::to_bytes(body),
    )
    .await
});
```

Reply:
Great suggestion.  Fixed.